queues.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+// A queue must be read and write
+type Queue interface {
+	send(DomainEntry) error
+	pull(int) ([]DomainEntry, error)
+	peek(string) int
+	persist() error
+}
+
+type cachedQueue struct {
+	store   map[string]int
+	backend Backend
+}
+
+func NewCachedQueue(persistanceBackend Backend) *cachedQueue {
+	q := new(cachedQueue)
+	q.store = make(map[string]int)
+	q.backend = persistanceBackend
+	return q
+}
+
+func (q cachedQueue) send(entry DomainEntry) error {
+	stats := q.store[entry.Domain]
+	if stats == -1 {
+		return nil
+	} else if !entry.IsBounce && stats >= 0 {
+		if stats >= 1000 { // a magic number
```

Comment:
What if it never get to reach this number and the process gets restarted for whatever reason?

Reply:
Aye! That is a problem. There is a TODO regarding adding persistence as a worker step. There is a stub for it right now in https://github.com/mailgun/catchall/pull/2/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R55

Reply:
Any other way to solve it without persistence?

Reply:
Without putting the values into durable storage I do not know how to save those entries. While a planned or graceful exit would be the most likely thing to happen and saving the info then would be possible, that situation isn't exactly what I think you are referring to.

This value is essentially batching all the counter synchronization calls to cassandra into sizes of N and does have a dramatic effect on performance. 

It could have a secondary node that it mirrors to ala replication.

Reply:
Just to clarify that when I meant persist regarding this situation I meant it would write to local storage (file, sqlite, etc).

I have definitely overloaded the word persist to also mean writing to cassandra. My apologies if this is a source of confusion.